CI is broken for faiss tests on Windows: node down: Not properly terminated

Faiss tests on Windows make the CI run indefinitely until maximum execution time is reached.

See: https://github.com/huggingface/datasets/actions/runs/9712659783
```
test (integration, windows-latest, deps-minimum)
The job running on runner GitHub Actions 60 has exceeded the maximum execution time of 360 minutes.
test (integration, windows-latest, deps-latest)
The job running on runner GitHub Actions 238 has exceeded the maximum execution time of 360 minutes.
```
```
____________________________ tests/test_search.py _____________________________
[gw1] win32 -- Python 3.8.10 C:\hostedtoolcache\windows\Python\3.8.10\x64\python.exe
worker 'gw1' crashed while running 'tests/test_search.py::IndexableDatasetTest::test_add_faiss_index'
____________________________ tests/test_search.py _____________________________
[gw2] win32 -- Python 3.8.10 C:\hostedtoolcache\windows\Python\3.8.10\x64\python.exe
worker 'gw2' crashed while running 'tests/test_search.py::IndexableDatasetTest::test_add_faiss_index'
```
```
tests/test_search.py::IndexableDatasetTest::test_add_faiss_index 
[gw0] node down: Not properly terminated
[gw0] FAILED tests/test_search.py::IndexableDatasetTest::test_add_faiss_index 

replacing crashed worker gw0

tests/test_search.py::IndexableDatasetTest::test_add_faiss_index 
[gw1] node down: Not properly terminated
[gw1] FAILED tests/test_search.py::IndexableDatasetTest::test_add_faiss_index 

replacing crashed worker gw1

tests/test_search.py::IndexableDatasetTest::test_add_faiss_index 
[gw2] node down: Not properly terminated
[gw2] FAILED tests/test_search.py::IndexableDatasetTest::test_add_faiss_index 

replacing crashed worker gw2
```

--- a/tests/utils.py
+++ b/tests/utils.py
@@ -67,7 +67,7 @@ require_not_windows = pytest.mark.skipif(
 )
 
 
-require_faiss = pytest.mark.skipif(find_spec("faiss") is None, reason="test requires faiss")
+require_faiss = pytest.mark.skipif(find_spec("faiss") is None or sys.platform == "win32", reason="test requires faiss")
 
 
 def require_regex(test_case):

--- a/tests/utils.py
+++ b/tests/utils.py
@@ -67,18 +67,7 @@ require_not_windows = pytest.mark.skipif(
 )
 
 
-def require_faiss(test_case):
-    """
-    Decorator marking a test that requires Faiss.
-
-    These tests are skipped when Faiss isn't installed.
-
-    """
-    try:
-        import faiss  # noqa
-    except ImportError:
-        test_case = unittest.skip("test requires faiss")(test_case)
-    return test_case
+require_faiss = pytest.mark.skipif(find_spec("faiss") is None, reason="test requires faiss")
 
 
 def require_regex(test_case):

